src/main/java/com/example/springapi/controller/MyRestController.java

```diff
@@ -1,93 +1,100 @@
+    String reversedJoke = jokeService.getReversedJoke(joke);
+    String jokeLength = jokeService.getJokeLength(joke);
+    return new ResponseEntity<>( joke + reversedJoke + jokeLength, HttpStatus.OK);
   }
 
-  //TODO Add constructor initialization
-  //TODO Remove internal joke object in classes
-  //TODO Save joke to Postgres database
-  //TODO Hibernate + Spring JPA
-  //TODO NonNull lombok
-  //TODO Sequence diagram
-  // Optional: add logger slf4j
-
-  @RequestMapping("/joke")
-  public String getJoke() {
-    String message = restTemplate.getForObject(URL_API, String.class);
-    LOGGER.debug(message);
-    String result = "";
-    ObjectMapper mapper = new ObjectMapper();
-    try {
-      Joke joke = mapper.readValue(message, Joke.class);
-      String jokeString = joke.toString();
-      LOGGER.debug(jokeString);
-      result = jokeString;
-      String jokeLen = jokeLength.actionWithJoke(joke);
-      LOGGER.debug(jokeLen);
-      result += "\n" + jokeLen;
-      String jokeRev = reverseJoke.actionWithJoke(joke);
-      LOGGER.debug(jokeRev);
-      result += "\n" + jokeRev;
-      jokeService.saveJoke(joke);
-    } catch (JsonProcessingException e) {
-      LOGGER.error(e.getMessage(), e);
+  @RequestMapping("/jokebase/typed/{subpath}")
+  public ResponseEntity<List<JokeEntity>> getJokesByType(@PathVariable("subpath") String subpath) {
+    List<JokeEntity> jokeEntities = new ArrayList<>(jokeService.findJokeByType(subpath));
+    if (jokeEntities.isEmpty()) {
+      return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
-    return result;
+    return new ResponseEntity<>(jokeEntities, HttpStatus.OK);
   }
 
   @RequestMapping("/jokebase")
-  public ResponseEntity<List<Joke>> getAllJokes() {
-    List<Joke> jokes;
-    jokes = new ArrayList<>(jokeService.findAll());
-
-    if (jokes.isEmpty()) {
+  public ResponseEntity<List<JokeEntity>> getAllJokes() {
+    List<JokeEntity> jokeEntities;
+    jokeEntities = new ArrayList<>(jokeService.getAllJokes());
+    if (jokeEntities.isEmpty()) {
       return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
 
-    return new ResponseEntity<>(jokes, HttpStatus.OK);
+    return new ResponseEntity<>(jokeEntities, HttpStatus.OK);
   }
 
   @RequestMapping("/jokebase/{subpath}")
-  public Joke getJokeById(@PathVariable("subpath") String subpath) {
+  public ResponseEntity<JokeEntity>  getJokeById(@PathVariable("subpath") String subpath) {
+    int id = 0;
+    JokeEntity jokeEntity;
+    try {
+      id = Integer.parseInt(subpath);
+    } catch (NumberFormatException e) {
+      log.error(e.getMessage(), e);
+    }
+    jokeEntity = jokeService.findJokeById(id);
+    if (jokeEntity == null) {
+      return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    } else {
+      return new ResponseEntity<>(jokeEntity, HttpStatus.OK);
+    }
+  }
+
+  @RequestMapping("/jokebase/length/{subpath}")
+  public ResponseEntity<String>  getLengthJokeById(@PathVariable("subpath") String subpath) {

```

Comment:
I would say Joke have length, and not the opposite
either you can write JokeLength or LengthOfJoke
I prefer 1st option

Reply:
done